etchosts/etchosts.go

```diff
@@ -119,25 +119,34 @@ func Add(path string, recs []Record) error {
 		return nil
 	}
 
-	f, err := os.Open(path)
+	b, err := mergeRecords(path, recs)
```

Comment:
To stick to the fd leak fix, I would we just `defer f.Close()` if no err here and keep the refactor for future enhancement.